fastar/core.py

```diff
@@ -130,7 +130,7 @@ def _stack(*args):
           else jnp.stack([sup if a is null else a for a in args]))
 
 def compress(updates):
-  static = [u and (u.meta_static, u.inshapes) for u in updates]
+  static = [u and (u.meta_static and tuple(sorted(u.meta_static.items())), u.inshapes) for u in updates]
```

Comment:
I think you need to change
```python
u.meta_static and tuple(sorted(u.meta_static.items()))
```
to
```python
None if u.meta_static is None else tuple(sorted(u.meta_static.items()))
```
so that emtpy dicts don't slip through.

For safety you could add the line
```python
assert u is None or u.meta_static is None or isinstance(u.meta_static, dict)
```
just above.